validator/remote/src/main/java/tech/pegasys/teku/validator/remote/FailoverValidatorApiHandler.java

```diff
@@ -254,12 +261,14 @@ public SafeFuture<List<SubmitDataError>> sendAggregateAndProofs(
   public SafeFuture<SendSignedBlockResult> sendSignedBlock(
       final SignedBlockContainer blockContainer,
       final BroadcastValidationLevel broadcastValidationLevel) {
-    final UInt64 slot = blockContainer.getSlot();
-    if (blockContainer.isBlinded() && blindedBlockCreatorCache.containsKey(slot)) {
-      final ValidatorApiChannel blockCreatorApiChannel = blindedBlockCreatorCache.remove(slot);
+    final SlotAndBlockRoot slotAndBlockRoot = blockContainer.getSignedBlock().getSlotAndBlockRoot();
+    if (blockContainer.isBlinded() && blindedBlockCreatorCache.containsKey(slotAndBlockRoot)) {
+      final ValidatorApiChannel blockCreatorApiChannel =
+          blindedBlockCreatorCache.remove(slotAndBlockRoot);
       LOG.info(
-          "Block for slot {} was blinded and will only be sent to the beacon node which created it.",
-          slot);
+          "Block for slot {} and block {} was blinded and will only be sent to the beacon node which created it.",
```

Comment:
We mention block twice, I think it's cleaner to say `Block for slot {} and root {}`